plugin/pkg/global/deletionconfirmation/admission.go

```diff
@@ -174,6 +174,14 @@ func (d *DeletionConfirmation) Validate(ctx context.Context, a admission.Attribu
 			if shootIgnored(obj) {
 				return fmt.Errorf("cannot delete shoot if %s annotation is set", v1beta1constants.ShootIgnore)
 			}
+			shoot, ok := obj.(*core.Shoot)
```

Comment:
Still, I'm asking myself, if the `ShootValidator` plugin wouldn't be the better place to put this check.
For example, it already checks the last operation for preventing finalizer removal for not successfully deleted shoots: https://github.com/gardener/gardener/blob/da556e69b265796e7b31f07b58f0b77ef66553c4/plugin/pkg/shoot/validator/admission.go#L298-L311

I think your new check is pretty similar and should be moved there.

Reply:
+1
I think the benefit is that we don't reject the `k delete` request but instead delete the shoot once it's safe to do so. Sounds user friendlier to me.

Reply:
I would also vote to keep `DeletionConfirmation` admission plugin focused only on deletion confirmation annotation check. Additional checks other than the deletion confirmation can go into `ShootValidator`.

Reply:
Ok, I will extend the `ShootValidator` to accept `Deletion` requests and I will move the logic there. My only concern is that we are splitting the `delete` admission / validation to two different places.